aeron-client/src/main/c/util/aeron_fileutil.c

```diff
@@ -453,6 +453,51 @@ int aeron_open_file_rw(const char *path)
 }
 #endif
 
+int aeron_mkdir_recursive(const char *pathname, int permission)
+{
+    if (aeron_mkdir(pathname, permission) == 0)
+    {
+        return 0;
+    }
+
+    if (errno != ENOENT)
+    {
+        return -1;
+    }
+
+    char *_pathname = strdup(pathname);
+    char *p;
+    int rc = -1;
+
+    for (p = _pathname + strlen(_pathname) - 1; p != _pathname; p--)
+    {
+        if (*p == AERON_FILE_SEP)
+        {
+            *p = '\0';
+            // _pathname is now the parent directory of the original pathname
+            rc = aeron_mkdir_recursive(_pathname, permission);
```

Comment:
We need to be a little bit smart about the error here.  If this returns -1, then we should be using AERON_APPEND_ERR to build the error stack.